resources/persist/src/lib.rs

```diff
@@ -25,22 +33,70 @@ pub enum PersistError {
 #[derive(Serialize)]
 pub struct Persist;
 
-#[derive(Deserialize, Serialize, Clone)]
+#[derive(Debug, Deserialize, Serialize, Clone)]
 pub struct PersistInstance {
     service_name: ServiceName,
 }
 
 impl PersistInstance {
-    pub fn save<T: Serialize>(&self, key: &str, struc: T) -> Result<(), PersistError> {
-        let storage_folder = self.get_storage_folder();
+    /// new method constructs a new PersistInstance along with its associated storage folder
+    pub fn new(service_name: ServiceName) -> Result<Self, PersistError<'static>> {
+        let instance = Self { service_name };
+        let storage_folder = instance.get_storage_folder();
         fs::create_dir_all(storage_folder).map_err(PersistError::CreateFolder)?;
 
+        Ok(instance)
+    }
+
+    pub fn save<T: Serialize>(&self, key: &str, struc: T) -> Result<(), PersistError> {
         let file_path = self.get_storage_file(key);
         let file = File::create(file_path).map_err(PersistError::Open)?;
         let mut writer = BufWriter::new(file);
         Ok(serialize_into(&mut writer, &struc).map_err(PersistError::Serialize))?
     }
 
+    /// list method returns a vector of strings containing all the keys associated with a PersistInstance
```

Comment:
Nit: should we capitalize these doc comments like we do elsewhere in the codebase? I also don't think we need "list method" as the first part of the comment.
```suggestion
    /// Returns a vector of strings containing all the keys associated with a [PersistInstance]
```